app_dart/test/service/luci_build_service_test.dart

```diff
@@ -0,0 +1,311 @@
+                builds: <Build>[linuxBuild],
+              ),
+            ),
+          ],
+        );
+      });
+      final Map<String, Build> builds =
+          await service.buildsForRepositoryAndPr('cocoon', 1, 'abcd');
+      expect(builds,
+          equals(<String, Build>{'Mac': macBuild, 'Linux': linuxBuild}));
+    });
+  });
+  group('scheduleBuilds', () {
+    setUp(() {
+      serviceAccountInfo = const ServiceAccountInfo(email: 'abc@abcd.com');
+      config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
+      mockBuildBucketClient = MockBuildBucketClient();
+      service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+    });
+    test('try to schedule builds already started', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+              searchBuilds: SearchBuildsResponse(
+                builds: <Build>[
+                  Build(
+                    id: 998,
+                    builderId: BuilderId(
+                      project: 'flutter',
+                      bucket: 'prod',
+                      builder: 'Linux',
+                    ),
+                    status: Status.started,
+                  )
+                ],
+              ),
+            ),
+          ],
+        );
+      });
+      final bool result = await service.scheduleBuilds(
+        prNumber: 1,
+        commitSha: 'abc',
+        repositoryName: 'cocoon',
+      );
+      expect(result, isFalse);
+    });
+    test('try to schedule builds already scheduled', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+              searchBuilds: SearchBuildsResponse(
+                builds: <Build>[
+                  Build(
+                    id: 998,
+                    builderId: BuilderId(
+                      project: 'flutter',
+                      bucket: 'prod',
+                      builder: 'Linux',
+                    ),
+                    status: Status.scheduled,
+                  )
+                ],
+              ),
+            ),
+          ],
+        );
+      });
+      final bool result = await service.scheduleBuilds(
+        prNumber: 1,
+        commitSha: 'abc',
+        repositoryName: 'cocoon',
+      );
+      expect(result, isFalse);
+    });
+    test('Schedule builds when the current list of builds is empty', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[],
+        );
+      });
+      config.luciTryBuildersValue =
+          (json.decode('[{"name": "Cocoon", "repo": "cocoon"}]')
+                  as List<dynamic>)
+              .cast<Map<String, dynamic>>();
```

Comment:
I guess we can simplify this to `json.decode(...) as Map<String, dynamic>`.

Reply:
This is actually a List of Map<String, dynamic> for some reason it requires the casting to happen in two steps.